### DISCUSSION TITLE
Should the one-way tabyl column "percent" be renamed?

### DISCUSSION THREAD

**Discussion by sfirke** [sfirke/janitor#474] [Ideas]
People keep complaining that the column name "percent" is technically inaccurate, see #300 and other related issues.  Should it be renamed?  I had figured it wasn't worth a breaking change, but maybe it is?  In practice I don't see how anyone is going to be confused, but if users are always renaming the column because they find it wrong, that's not great.

If it should be renamed, what is the best replacement name?

**Comment by billdenney** [comment]
I think that breaking changes should make it clear to the user that things are broken-- even if they aren't paying attention.  For that, I think that the simplest change would be to rename the column to "fraction".  That way, if they have code looking for "percent", it will break and they'll find it quickly.

I think it would be harder if the name stays "percent" and the value changes to be 100-times the current number.  Then, code would break but in a silent way.  The way to do that, in my opinion, would be to add an argument like `percent_max=1` and the user could change it to 100 (or any other number...).

Or a simple way (for the user and not for the maintainer) is to borrow the concept of editions from other tidyverse packages (I first think of `testthat`).  And, if you're using the new edition, you get a percent.  If you're using the old edition, you get the fraction.

**Reply by sfirke** [reply]
Good ideas, I like keeping the functionality the same but changing the name to `fraction` or `prop`.
I hadn't seen editions but I just studied how it works for testthat.  Elegant approach though it seems more geared to package developers and most people use janitor interactively.  I am lucky that there aren't 5k packages depending on janitor 😅

**Comment by jzadra** [comment]
I agree with @billdenney that changing the value without changing the name is a bad idea in the current package.

@sfirke You've talked before about separating the cleaning functions of janitor and the data summarization functions to create a second package for the latter. Perhaps rather than making a new edition, this change could be implemented when this new package is created.

**Reply by sfirke** [reply]
My first reaction to this was oof, I'm never gonna get that done.

But maybe I don't have to write any code, and this would be the time?  I could take an afternoon or weekend and split everything up?  I kind of like this.  I can't get to it soon but this change is also not time-sensitive, and if I need/want to release another version of janitor in the meantime I can.

**Comment by billdenney** [comment]
A split does sound like the cleanest path.

For that, perhaps something like the following would make sense:

* Split the code to a new `tabyl` package.
* Add a `packageStartupMessage()` to janitor telling users about it and the fact that `tabyl`s have a different column instead of `percent` (`fraction`?).
* Only use the new column in the `tabyl` package without backward compatibility-- and put that in the news file.

From there, hopefully, it's only a few revdep checks for the CRAN packages (https://devtools.r-lib.org/reference/revdep.html).

**Comment by dhduncan** [comment]
I am one of the complainers. I mean, I am the type to complain, though I came to check the issues log first and found my way here. I have just started using `janitor` after first noticing it some years ago.  The enhancements in `tabyl` compared to `table` are valuable to me, but the mislabelling of the output does seem like something that it would be good to fix.  I would favour "proportion", or "prop" over "fraction" or "share", mainly on vibes.

If I had package management smarts I could offer to help. Or am I coding in the past?
